app/models/course/enrol_request.rb

```diff
@@ -0,0 +1,27 @@
+class Course::EnrolRequest < ActiveRecord::Base
+  stampable
+  has_paper_trail on: [:destroy]
+  validates_uniqueness_of :course_id, scope: :user_id
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.